### DISCUSSION TITLE
MFA?

### DISCUSSION THREAD

**Discussion by kylestewart** [PowerShell/Win32-OpenSSH#2047] [Q&A]
Hello!

I'm trying to figure out whether I would be able to add a custom MFA solution to this.  Is there PAM support built-in? Or another way we can do it?

**Comment by tgauth** [comment]
PAM is unsupported at this time, unfortunately!

**Reply by kylestewart** [reply]
Ah, bummer!  Thank you for your response!

**Comment by NoMoreFood** [comment]
If you're talking about a hardware token, you can use OpenSSH's PKCS or FIDO support.  Those work on the Windows client.  You could also use OpenSSH for Windows Server on the backend and something like PuTTY CAC if you're looking for something more "user friendly".

**Reply by kylestewart** [reply]
Good advice, thank you!
